CRM/Contribute/BAO/Contribution.php

```diff
@@ -4722,7 +4722,7 @@ public static function sendMail(&$input, &$ids, $contributionID, &$values,
     $values['contribution_status'] = CRM_Core_PseudoConstant::getLabel('CRM_Contribute_BAO_Contribution', 'contribution_status_id', $contribution->contribution_status_id);
     $return = $contribution->composeMessageArray($input, $ids, $values, $returnMessageText);
     // Contribution ID should really always be set. But ?
```

Comment:
This comment doesn't make sense to me. Maybe it's no longer relevant?

Reply:
Comment is obsolete - on line 4714 the existence of the contribution is tested, and this test fails when the id is not set.